### PR TITLE
feat(clerk-js): Redesign Connected Accounts, Web3 and MFA in <UserProfile/>

### PR DESCRIPTION
**Pull request by octoper**
## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-p9mw58l08.clerkstage.dev) | Jan 05, 2024 11:27 AM |

**Review by anagstef — APPROVED** [review]

File: packages/clerk-js/src/ui/elements/Menu.tsx [hunk 1]
```diff
@@ -17,24 +19,31 @@ type MenuState = {
...
+    placement: popoverPlacement,
     offset: 8,
     bubbles: false,
+    shoudFlip: false,
   });
 
-  const value = React.useMemo(() => ({ value: { popoverCtx, elementId } }), [{ ...popoverCtx }, elementId]);
+  const value = React.useMemo(
+    () => ({ value: { popoverCtx, elementId: rest.elementId } }),
+    [{ ...popoverCtx }, rest.elementId],
```
[the remark this answers is not in the document]
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Menu.tsx]: Good catch, fixed!

File: packages/clerk-js/src/ui/elements/Menu.tsx [hunk 2]
```diff
@@ -17,24 +19,28 @@ type MenuState = {
   elementId?: MenuId;
 };
 
-const [MenuStateCtx, useMenuState] = createContextAndHook<MenuState>('MenuState');
+export const [MenuStateCtx, useMenuState] = createContextAndHook<MenuState>('MenuState');
 
-type MenuProps = PropsWithChildren<Record<never, never>> & { elementId?: MenuId };
+type MenuProps = PropsWithChildren<Record<never, never>> & {
+  elementId?: any;
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Menu.tsx:25]: Why do we use `any` here?

Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Menu.tsx:25]: Why do we use `any` here instead of `MenuId` ?

File: packages/clerk-js/src/ui/elements/Menu.tsx [hunk 3]
```diff
@@ -17,24 +19,28 @@ type MenuState = {
...
+  popoverPlacement?: Placement;
+};
 
 export const Menu = withFloatingTree((props: MenuProps) => {
-  const { elementId } = props;
+  const { popoverPlacement = 'bottom-end', elementId, ...rest } = props;
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Menu.tsx:30]: Shouldn't we use `right-start` as a default like before?

File: packages/clerk-js/src/ui/elements/Menu.tsx [hunk 4]
```diff
@@ -72,10 +78,12 @@ const findMenuItem = (el: Element, siblingType: 'prev' | 'next', options?: { cou
   return sibling;
 };
 
-type MenuListProps = PropsOfComponent<typeof Col>;
+type MenuListProps = PropsOfComponent<typeof Col> & {
+  isPortal?: boolean;
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/Menu.tsx:82]: Why is this needed?

File: packages/clerk-js/src/ui/elements/Menu.tsx (hunk 2 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Menu.tsx:25]: This happened because the trigger of `<ProfileSectionActionMenu/>` needs to have the same descriptor as the `<ProfileSectionButton/>` which is `descriptors.profileSectionPrimaryButton` so we don't have any breaking changes for customers that already using that descriptor, the change of the type `elementId` now is not only `MenuId` so I changed it to accept any value, I wanted to have this as a string because that is what are we passing but Typescript didn't liked that.

File: packages/clerk-js/src/ui/elements/Menu.tsx (hunk 3 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Menu.tsx:30]: With `right-start` some place we where using the Menu like the actions on the `<UserProfile/> Email addresses
section `, the popover was floating outside the component, so by using `bottom-end` looked a bit nicer.

With `right-start`:
CleanShot 2024-01-05 at 21 37 47

With `bottom-end`:
CleanShot 2024-01-05 at 21 39 02

File: packages/clerk-js/src/ui/elements/Menu.tsx (hunk 4 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/Menu.tsx:82]: Before, we always had the `<Menu/>` popover rendering in a Portal. However, for the `<ProfileSectionActionMenu/>`, we wanted to limit the width of the popover to match the size of the menu trigger. I added that flag to provide as an option to opt-out of Portal in cases where it's not desired, like the one I mentioned.

File: packages/clerk-js/src/ui/elements/Menu.tsx (hunk 2 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/elements/Menu.tsx:25]: It's better to align the types. You can do `type MenuId = 'invitation' | 'member' | ProfileSectionId`

File: packages/clerk-js/src/ui/elements/Menu.tsx (hunk 4 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/elements/Menu.tsx:82]: Can we name it `portal` or `asPortal` ?

**Review by desiprisg — APPROVED** [review]
